src/vs/workbench/browser/parts/editor/editorCommands.ts

```diff
@@ -656,7 +656,7 @@ function registerFocusEditorGroupAtIndexCommands(): void {
 	}
 }
 
-export function splitEditor(editorGroupService: IEditorGroupsService, direction: GroupDirection, contexts?: (IEditorCommandsContext | URI | undefined)[]): void {
+export function splitEditor(editorGroupService: IEditorGroupsService, direction: GroupDirection, contexts?: (IEditorCommandsContext | URI)[]): void {
```

Comment:
Multi select tabs and use context menu to split

Reply:
No, I mean across groups. This does not seem possible at all today, even from open editors view, so I suggest to make this operate only on one group to simplify the change:

![image](https://github.com/microsoft/vscode/assets/900690/36269724-7968-4959-9701-3f2286a78664)

Reply:
Sorry for nitpicking, but can we then bring back this method to how it used to be before multi-select and to indicate that this method never supports multiple groups from its function declaration? Here is how this method was implemented:

```ts
export function splitEditor(editorGroupService: IEditorGroupsService, direction: GroupDirection, context?: IEditorCommandsContext): void {
	let sourceGroup: IEditorGroup | undefined;
	if (context && typeof context.groupId === 'number') {
		sourceGroup = editorGroupService.getGroup(context.groupId);
	} else {
		sourceGroup = editorGroupService.activeGroup;
	}

	if (!sourceGroup) {
		return;
	}

	// Add group
	const newGroup = editorGroupService.addGroup(sourceGroup, direction);

	// Split editor (if it can be split)
	let editorToCopy: EditorInput | undefined;
	if (context && typeof context.editorIndex === 'number') {
		editorToCopy = sourceGroup.getEditorByIndex(context.editorIndex);
	} else {
		editorToCopy = sourceGroup.activeEditor ?? undefined;
	}

	// Copy the editor to the new group, else create an empty group
	if (editorToCopy && !editorToCopy.hasCapability(EditorInputCapabilities.Singleton)) {
		sourceGroup.copyEditor(editorToCopy, newGroup, { preserveFocus: context?.preserveFocus });
	}

	// Focus
	newGroup.focus();
}
```

In `registerSplitEditorCommands` we can do the filtering to only take the first `IEditorCommandsContext`.

Reply:
We have to pass the array because we want to support splitting multiple editors from the same group.